GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -1304,6 +1311,25 @@ private void UpdatePlaceholders()
 
                     if (keepFolder)
                     {
+                        if (this.TryGetFolderDataFromTreeUsingPath(folderPlaceholder.Path, out FolderData folderData))
+                        {
+                            string newFolderSha = folderData.HashedFileShas();

```

Comment:
>HashedFileShas [](start = 61, length = 14)

Is this the hash of hashes of contents of files in that folder?

If yes, this means `.vsf_usn_folder` will be created if the content of a file in that folder changes. From BuildXL's point of view, this will force BuildXL to perform extra validation.

For example, suppose that a task only enumerates a folder, but does not read any file within that folder. BuildXL will track the fingerprint of that folder. The fingerprint is the hash of file names (along with some attributes -- i.e., if it's a file or a directory). If the content of any file in that folder changes, but there's no membership change, then BuildXL can safely assume that the fingerprint remains the same, and concludes that the task is still up-to-date. However, the introduction of `.vfs_usn_folder` triggers a membership change (due to creation and deletion). This forces BuildXL to re-validate if the new fingerprint is still the same.

Perhaps, instead of HashedFileShas, GVFS4Git should track the same form of fingerprint as BuildXL.